multi-agent/lib/mas/resources/errors.py

```
@@ -15,7 +15,50 @@ def __init__(self, *, by_blueprints: list[str], by_resources: list[str]):
         super().__init__(msg)
 
-    # optional: for `str(exc)` explicitness in logs
+    def __str__(self) -> str:
+        return self.args[0]
+
+    def __repr__(self) -> str:
+        return self.__str__()
+
+
+class BuiltInWriteProtectedError(RuntimeError):
+    def __init__(self):
```

Ruff (0.15.21)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)